realm-transformer/src/main/groovy/io/realm/transformer/BytecodeModifier.groovy

```diff
@@ -147,4 +149,15 @@ class BytecodeModifier {
                     new CtClass[0], new CtClass[0], "{return true;}", clazz))
         }
     }
+
+    private static CtMethod createSetter(String methodName, CtField field) {
+        CtMethod setter = CtNewMethod.setter(methodName, field)
+        if ("io.realm.RealmInteger".equals(field.getType().getName())) {
+            setter.insertBefore(
+                    'if (!($1 instanceof io.realm.internal.datatypes.realminteger.UnmanagedRealmInteger))' +
+                    'throw new IllegalStateException("Cannot assign a managed RealmInteger to an unmanaged object");');
```

Comment:
Ah, I overlooked `!`.

But why do we need to reject managed `RealmInteger`?
I don't think we should interfere unmanaged object. It's normal Java Object.
We should allow everything Java is allowing.